### PR TITLE
Updated out-dated doc pages to now use slashcommands.

### PR DESCRIPTION
**Pull request by Jaskowicz1**
Some doc pages were still using `bot.on_message_create` for commands, rather than using `bot.on_slashcommand`. This PR changes those pages to ensure that documentation is up-to-date and following discord's standards.

This PR also alters "Advanced components" (components2.md) to now be named "Advanced button components" as the current name doesn't show that the page is a follow up of "Using button components".

### CODE REVIEW

**Review by braindigitalis — APPROVED** [review]

**Comment by braindigitalis** [comment]
looks good to me!

however if we remove message commands from all examples we need to create one new example that shows how to detect content in messages as there will be no remaining examples of on_message_create.

**Comment by Jaskowicz1** [comment]
> looks good to me!
> 
> however if we remove message commands from all examples we need to create one new example that shows how to detect content in messages as there will be no remaining examples of on_message_create.

That sounds good, I wanted to add a couple new pages in a different PR so, if it's okay with you, after this PR i'll open a new PR for adding new pages (for example: on_message_create, moving listeners into separate classes, etc)

**Comment by Jaskowicz1** [comment]
This PR now includes an addition to slashcommands.md (added more information) along with a change to coding_style_standards (just to give more information about curly braces).

**Review by braindigitalis — APPROVED** [review]

**Comment by Mishura4** [comment]
Looks good! Just to make sure, did you test the examples you changed? They build and work?

**Comment by Jaskowicz1** [comment]
> Looks good! Just to make sure, did you test the examples you changed? They build and work?

Everything excluding the music/voice was tested as I do not have the dependencies required. If a test is wanted for those, I can get a test out tomorrow and report back with the results of that!

**Comment by Mishura4** [comment]
That'd be amazing 🫡

**Review by Commandserver — COMMENTED** [review]
cool

File: docpages/example_programs/music_and_audio/join_voice.md [hunk 1]
```diff
@@ -12,30 +12,28 @@ When a user issues a command you may want to join their voice channel, e.g. in a
...
 			bool join_vc = true;
-			/* Check if we are currently on any vc */
-			if (current_vc) {
-				/* Find the channel id  that the user is currently on */
-				auto users_vc = g->voice_members.find(event.msg.author.id);
-				/* See if we currently share a channel with the user */
+
+			if (!current_vc) {
+				auto users_vc = g->voice_members.find(event.command.get_issuing_user().id);
+
```
[the remark this answers is not in the document]
Comment (braindigitalis) [reply docpages/example_programs/music_and_audio/join_voice.md:33]: yes in this case the comments are very important as people kept assuming that connect member voice was synchronous

File: docpages/example_programs/misc/cache_messages.md [hunk 2]
```diff
@@ -14,44 +14,56 @@ exercise to the reader. For further reading please see the documentation of dpp:
...
 		/* Store the new pointer to the cache using the store() method */
 		message_cache.store(m);
+	});
+
+	/* The event is fired when someone issues your commands */
+	bot.on_slashcommand([&bot](const dpp::slashcommand_t& event) {
+		/* Check which command they ran */
+		if (event.command.get_command_name == "get") {
+
+			dpp::message* find_msg = message_cache.find(std::get<std::string>(event.get_parameter("message_id")));
```
[the remark this answers is not in the document]
Comment (Jaskowicz1) [reply docpages/example_programs/misc/cache_messages.md:41]: This does compile and work.

**Comment by Jaskowicz1** [comment]
Voice commands have been tested and are working as intended.

**Review by Mishura4 — APPROVED** [review]

**Pull request merged**